jwst/pipeline/calwebb_image3.py

```diff
@@ -69,22 +67,23 @@ def process(self, input_data):
 
         self.source_catalog.save_results = self.save_results
 
-        with datamodels.open(input_data, asn_exptypes=asn_exptypes) as input_models:
+        # Only load science members from input ASN;
+        # background and target-acq members are not needed.
+        input_models = self._load_input_as_library(input_data)
+
+        if (self.output_file is None) and hasattr(input_models.asn["products"][0], "name"):
             # If input is an association, set the output to the product name.
-            if self.output_file is None:
-                try:
-                    self.output_file = input_models.meta.asn_table.products[0].name
-                except (AttributeError, IndexError):
-                    pass
+            self.output_file = input_models.asn["products"][0]["name"]
 
-            # Check if input is single or multiple exposures
-            try:
-                has_groups = len(input_models.group_names) >= 1
-            except (AttributeError, TypeError, KeyError):
-                has_groups = False
+        # Check if input is single or multiple exposures
+        has_groups = len(input_models) > 1
```

Comment:
```suggestion
        has_groups = len(input_models.group_names) > 1
```
The old code skipped tweakreg (and some other stuff) if the input only had 1 group (which might contain >1 model).